### PR TITLE
Retry fulfillment even if the attempt was failed by HttpClientError

### PR DESCRIPTION
**Pull request by schenedx**
Need to retry fulfillment task even on HttpClientError.
@edx/ecommerce Please review

### CODE REVIEW

**Review by clintonb — APPROVED** [review]
Tests?

**Comment by MatthewPiatetsky** [comment]
I'm used to seeing retry behavior in python as a decorator, but I guess in this case, we don't want to retry in all paths for that function. 
👍

File: ecommerce_worker/fulfillment/v1/tasks.py [hunk 1]
```diff
@@ -10,6 +10,22 @@
...
+    """
+    Retry with exponential backoff until fulfillment
+    succeeds or the retry limit is reached. If the retry limit is exceeded,
+    the exception is re-raised.
+    """
+    retries = self.request.retries
+    if retries == max_fulfillment_retries:
+        logger.exception('Fulfillment of order [%s] failed. Giving up.', order_number)
+    else:
+        logger.warning('Fulfillment of order [%s] failed. Retrying.', order_number)
```
Comment (MatthewPiatetsky) [inline ecommerce_worker/fulfillment/v1/tasks.py:23]: Do we want to log any information about which retry we're on?

Comment (schenedx) [reply ecommerce_worker/fulfillment/v1/tasks.py:23]: I don't find that information to be valuable. We don't record down that information anywhere. It serves to fulfill an curiosity more than anything else I think

**Comment by schenedx** [comment]
@clintonb Test added.

**Pull request merged**